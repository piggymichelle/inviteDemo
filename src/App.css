.header {
 height: 60px;
 width:100%;
 border: 1px solid #000;
 background: #fff;
 display: flex;
 align-items: center;
 color: grey;
}
.headerContent {
  width: 80%;
  margin: 0 auto;
}
.container {
  display: flex;
  min-height: 100vh;
  flex-direction: column;
}
.mainContent {
  display: flex;
  flex: 1;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}
button{
  height: 40px;
  width:130px;
  border-radius: 2px;
  border: 1px solid #000;
  background-color:#fff;
  color: grey;
  box-shadow: 0px 8px 15px rgba(148, 132, 132, 0.1);
  cursor: pointer;
  transition: all 0.6s;
}
button:hover {
  background-color: rgb(244, 247, 220);
  box-shadow: 0px 8px 15px rgba(148, 132, 132, 0.1);
}

.footer {
  height: 60px;
  bottom:0;
  border: 1px solid #000;
  font-size: 0.4rem;
  color: grey;
  background: #fff;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}
.slogan {
  font-size: 2.5rem;
  color: grey;
  display: flex;
  flex-direction: column;
  align-items: center;
}
p {
  font-size: 0.8rem;
  color: grey;
}
