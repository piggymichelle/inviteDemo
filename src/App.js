import React from 'react';
import './App.css';
import Content from './mainContent'
function App() {
  return (
    <div>
      <Content/>
    </div>
  );
}

export default App;
