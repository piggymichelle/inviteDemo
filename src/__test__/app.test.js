import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';
import Content from '../mainContent'
import Modal from '../Modal'
import { shallow } from 'enzyme';
import { wrap } from 'module';


describe('my app test cases', ()=>{
  //app test

  it('should renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div)
  });
  
  it('should render content component correctly', () => {
    const wrapper = shallow(<App />)
    expect(wrapper.find(Content).exists()).toEqual(true)
  });

  it('snapshot test',()=>{
    const wrapper = shallow(<App />)
    expect(wrapper).toMatchSnapshot()
  })
  //modal test
  it('mainContent state should initial correct',()=>{
    const wrapper = shallow(<Content />)
    const state = {
        isOpenModal:false
      }
    expect(wrapper.state()).toEqual(state)
  })

  it('mainContent click function should be invoke correctly',()=>{
    const wrapper = shallow(<Content />)
    const spy = jest.spyOn(wrapper.instance(), 'openModal')
    wrapper.instance().forceUpdate()
    wrapper.find('button').simulate('click')
    expect(spy).toHaveBeenCalled()
  })

  it('mainContent click function should be invoke correctly',()=>{
    const wrapper = shallow(<Content />)
    const currState = wrapper.state().isOpenModal
    wrapper.find('button').simulate('click')
    expect(wrapper.state().isOpenModal).toEqual(!currState)
  })

  it('modal state should initial correct',()=>{
    const wrapper = shallow(<Modal />)
    const state = {
        fullName:'', 
        email: '', 
        confEmail: null, 
        isSending:false, 
        success:false, 
        invalid:false,
        data:''
      }
    expect(wrapper.state()).toEqual(state)
  })

  it('modal rendering correctly',()=>{
    const props = {show:true}
    const wrapper = shallow(<Modal {...props}/>)
      expect(wrapper.find('.backdrop').exists()).toEqual(true)
  })

  it('form rendering correctly',()=>{
    const props = {show:true}
    const wrapper = shallow(<Modal {...props}/>)
      expect(wrapper.find('form').exists()).toEqual(true)
  })

  it('submit function should be called',()=>{
    const props = {show:true}
    const wrapper = shallow(<Modal {...props}/>)
      const spy = jest.spyOn(wrapper.instance(), 'handleSubmit')
      wrapper.instance().forceUpdate()
      wrapper.find('button').simulate('click',{ preventDefault() {} })
      expect(spy).toHaveBeenCalled()
  })
})

