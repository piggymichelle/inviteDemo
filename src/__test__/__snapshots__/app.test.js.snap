// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my app test cases snapshot test 1`] = `ShallowWrapper {}`;
